scraper.rb

```diff
@@ -36,17 +37,18 @@ def remove_prefixes(name)
 def scrape_area(url)
   noko = noko_for(url)
   area_id, area = noko.css('h2').text.tidy.match(/Constituency (\d+), (.*)/).captures
-  noko.css('.article-content tr').each do |tr|
-    tds = tr.css('td') 
-    next unless tds[0] && tds[0].text.match(/\d/)
-    pref, name = remove_prefixes(tds[1].text.tidy.sub('Col.X','Col X'))
+  area = area.match(/[a-zA-Z ]+/).to_s
+  noko.css('.article-content p').children.each do |row|
+    next unless row.class == Nokogiri::XML::Text
```

Comment:
Again, _if_ you need to do this, it would be better skipping the ones that are empty, rather than explicitly checking for a class. But, again, I suspect you'd be much better simply selecting the correct elements in the first place instead.

Perhaps `noko.xpath('//div[@class="article-content"]//p//text()')`?